File: android_samples_java/02 App_basic_CallBackExamples/app/src/main/java/com/msnegi/callbackexamples/CallbackFragment.java
package com.msnegi.callbackexamples;

import android.content.Context;
import android.os.Bundle;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

public class CallbackFragment extends Fragment {

    CallBackInterface callBackInterface;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

        try{
            callBackInterface = (MainActivity) context;

        }catch (ClassCastException e){
            throw new ClassCastException(context.toString() + " must implement OnListItemSelectedListener");
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
    {
        View view = inflater.inflate(R.layout.fragment_callback, container, false);

        Button btnCallback = (Button) view.findViewById(R.id.btnCallback);
        btnCallback.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                callBackInterface.callbackfunction("This is a callback from Fragment !!!");
            }
        });

        return view;

    }

}

File: android_samples_kotlin/12 CallbackExamples/app/src/main/java/com/msnegi/callbackexamples/CallbackFragment.kt
package com.msnegi.callbackexamples

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.fragment.app.Fragment

class CallbackFragment : Fragment() {
    var callBackInterface: CallBackInterface? = null

    override fun onAttach(context: Context) {
        super.onAttach(context)

        try {
            callBackInterface = context as MainActivity
        } catch (e: ClassCastException) {
            throw ClassCastException(context.toString() + " must implement OnListItemSelectedListener")
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        val view = inflater.inflate(R.layout.fragment_callback, container, false)

        val btnCallback = view.findViewById<View?>(R.id.btnCallback) as Button
        btnCallback.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                callBackInterface!!.callbackfunction("This is a callback from Fragment !!!")
            }
        })

        return view
    }
}
